client/components/LearnerViewDirective.js

```diff
@@ -733,6 +739,33 @@ tie.directive('learnerView', [function() {
         };
 
         /**
+          * Highlights syntax errors in the coding UI
+          *
+          * @param {number} lineNumber
+          */
+        var highlightLine = function(lineNumber) {
+          var actualLineNumber = lineNumber - 1;
+          var codeLines = document.querySelectorAll('.CodeMirror-line');
+          if ($scope.isInDarkMode === true) {
+            codeLines[actualLineNumber].className +=
+              ' tie-syntax-error-line night-mode';
```

Comment:
Hm. So if I change the mode midway, what happens?

I think this function shouldn't reference night-mode at all. Instead, there should be a CSS rule whose selector is something like `.parent-element.night-mode .tie-syntax-error-line`.